cloudwatch_logs_common/src/log_publisher.cpp

```diff
@@ -108,102 +107,137 @@ Aws::CloudWatchLogs::ROSCloudWatchLogsErrors LogPublisher::StopPublisherThread()
   return CW_LOGS_SUCCEEDED;
 }
 
-void LogPublisher::Run()
+void LogPublisher::CreateGroup()
 {
-  Aws::CloudWatchLogs::ROSCloudWatchLogsErrors publisher_status = CW_LOGS_SUCCEEDED;
-  Aws::String next_token;
-  while (this->thread_keep_running_.load(std::memory_order_acquire)) {
-    // if log group cannot be created or log stream doesnt already exist, don't try to send to CW
-    if (!this->does_group_exist_) {
-      Aws::CloudWatchLogs::ROSCloudWatchLogsErrors check_log_group_exists_status =
-        this->cloudwatch_facade_->CheckLogGroupExists(this->log_group_);
-      if (CW_LOGS_SUCCEEDED == check_log_group_exists_status) {
-        this->does_group_exist_ = true;
-        AWS_LOGSTREAM_DEBUG(__func__, "Found existing log group: " << log_group_);
-        continue;
-      }
-      Aws::CloudWatchLogs::ROSCloudWatchLogsErrors create_log_group_status =
-        this->cloudwatch_facade_->CreateLogGroup(this->log_group_);
-      if (CW_LOGS_SUCCEEDED == create_log_group_status) {
-        this->does_group_exist_ = true;
-        AWS_LOG_DEBUG(__func__, "Successfully created log group.");
-      } else if (CW_LOGS_LOG_GROUP_ALREADY_EXISTS == create_log_group_status) {
-        this->does_group_exist_ = true;
-        AWS_LOG_DEBUG(__func__, "Log group already exists, quit attempting to create it.");
-      } else {
-        AWS_LOG_ERROR(__func__, "Failed to create log group, retrying ...");
-        continue;
-      }
-    }
+  Aws::CloudWatchLogs::ROSCloudWatchLogsErrors check_log_group_exists_status =
+    this->cloudwatch_facade_->CheckLogGroupExists(this->log_group_);
+  if (CW_LOGS_SUCCEEDED == check_log_group_exists_status) {
+    this->run_state_ = Aws::CloudWatchLogs::LOG_PUBLISHER_RUN_CREATE_STREAM;
+    AWS_LOGSTREAM_DEBUG(__func__, "Found existing log group: " << log_group_);
+    return;
+  }
+  Aws::CloudWatchLogs::ROSCloudWatchLogsErrors create_log_group_status =
+    this->cloudwatch_facade_->CreateLogGroup(this->log_group_);
+  if (CW_LOGS_SUCCEEDED == create_log_group_status) {
+    this->run_state_ = Aws::CloudWatchLogs::LOG_PUBLISHER_RUN_CREATE_STREAM;
+    AWS_LOG_DEBUG(__func__, "Successfully created log group.");
+  } else if (CW_LOGS_LOG_GROUP_ALREADY_EXISTS == create_log_group_status) {
+    this->run_state_ = Aws::CloudWatchLogs::LOG_PUBLISHER_RUN_CREATE_STREAM;
+    AWS_LOG_DEBUG(__func__, "Log group already exists, quit attempting to create it.");
+  } else {
+    AWS_LOG_ERROR(__func__, "Failed to create log group, retrying ...");
```

Comment:
Could you log out the error recieved here

Reply:
Done.